### PR TITLE
CODENVY-191: Add update factory client impl

### PR DESCRIPTION
**Pull request by akorneta**

### CODE REVIEW

File: core/platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java [hunk 1]
```diff
@@ -357,6 +357,26 @@ void push(String workspaceId,
...
+     *         project
+     * @param refSpec
+     *         list of refspec to push
+     * @param remote
+     *         remote repository name or url
+     * @param force
+     *         push refuses to update a remote ref that is not an ancestor of the local ref used to overwrite it. If <code>true</code>
+     *         disables the check. This can cause the remote repository to lose commits
+     */
+    Promise<PushResponse> push(String wsId,
```
Comment (sleshchenko) [inline core/platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java:378]: please, mark old method as deprecated

Comment (akorneta) [inline core/platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java:378]: done

**Comment by skabashnyuk** [comment]
ok

File: core/platform-api-client-gwt/che-core-client-gwt-factory/src/main/java/org/eclipse/che/api/factory/gwt/client/FactoryServiceClient.java [hunk 2]
```diff
@@ -90,4 +90,7 @@
      * @return a promise that will provide a list of {@link Factory}s, or rejects with an error
      */
     Promise<List<Factory>> findFactory(Integer skipCount, Integer maxItems, List<Pair<String, String>> params);
+
+
+    Promise<Factory> updateFactory(String id, Factory factory);
```
Comment (vzhukovs) [inline core/platform-api-client-gwt/che-core-client-gwt-factory/src/main/java/org/eclipse/che/api/factory/gwt/client/FactoryServiceClient.java:103]: Javadoc for the api method?

File: core/platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java (hunk 1 above)
Comment (vzhukovs) [inline core/platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java:378]: You use only string path from the `ProjectConfigDto`. So there is no need to pass whole configuration into this method.

File: plugins/plugin-github/che-plugin-github-ext-github/src/main/java/org/eclipse/che/ide/ext/github/client/GitHubClientService.java [hunk 3]
```diff
@@ -275,4 +275,9 @@ void createPullRequest(@NotNull String user,
      *         callback called when operation is done.
      */
     void updatePublicKey(@NotNull AsyncRequestCallback<Void> callback);
-}
\ No newline at end of file
+
+    Promise<GitHubPullRequest> updatePullRequest(String user,
```
Comment (vzhukovs) [inline plugins/plugin-github/che-plugin-github-ext-github/src/main/java/org/eclipse/che/ide/ext/github/client/GitHubClientService.java:292]: Javadoc for the api method?

File: core/platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java (hunk 1 above)
Comment (akorneta) [inline core/platform-api-client-gwt/che-core-client-gwt-git/src/main/java/org/eclipse/che/api/git/gwt/client/GitServiceClient.java:378]: Also it should be fixed over other issue, not related to this, here I only rewrite methods with the promises without changing in the common signature.

**Pull request merged**